framework/test/unit/modules/dpos_v2/auxiliary.spec.ts

```diff
@@ -0,0 +1,570 @@
+import { StateStore } from '@liskhq/lisk-chain';
+import { getRandomBytes } from '@liskhq/lisk-cryptography';
+import { InMemoryKVStore } from '@liskhq/lisk-db';
+import { when } from 'jest-when';
+import { updateDelegateProductivity } from '../../../../src/modules/dpos_v2/auxiliary';
+import {
+	MODULE_ID_DPOS,
+	STORE_PREFIX_DELEGATE,
+	STORE_PREFIX_PREVIOUS_TIMESTAMP,
+} from '../../../../src/modules/dpos_v2/constants';
+import {
+	delegateStoreSchema,
+	previousTimestampStoreSchema,
+} from '../../../../src/modules/dpos_v2/schemas';
+import { DelegateAccount } from '../../../../src/modules/dpos_v2/types';
+import { SubStore } from '../../../../src/node/state_machine/types';
+import { createBlockContext } from '../../../../src/testing';
+
+describe('DPoSAuxiliary', () => {
+	const EMPTY_KEY = Buffer.alloc(0);
+
+	const defaultConfigs = {
+		factorSelfVotes: 10,
+		maxLengthName: 20,
+		maxNumberSentVotes: 10,
+		maxNumberPendingUnlocks: 20,
+		failSafeMissedBlocks: 50,
+		failSafeInactiveWindow: 260000,
+		punishmentWindow: 780000,
+		roundLength: 103,
+		bftThreshold: 68,
+		minWeightStandby: BigInt(1000) * BigInt(10 ** 8),
+		numberActiveDelegates: 101,
+		numberStandbyDelegates: 2,
+		tokenIDDPoS: {
+			chainID: 0,
+			localID: 0,
+		},
+	};
+
+	let validatorsAPI: any;
+	let stateStore: StateStore;
+	let delegateDatas: DelegateAccount[];
+	let delegateAddresses: Buffer[];
+	let previousTimestampStore: SubStore;
+	let delegateStore: SubStore;
+	describe('updateDelegateProductivity', () => {
+		beforeEach(async () => {
+			stateStore = new StateStore(new InMemoryKVStore());
+
+			validatorsAPI = {
+				getGeneratorsBetweenTimestamps: jest.fn(),
+				getGeneratorAtTimestamp: jest.fn(),
+			};
+
+			delegateDatas = Array(103)
+				.fill({})
+				.map((_, index) => ({
+					name: `delegate${index}`,
+					totalVotesReceived: BigInt(0),
+					selfVotes: BigInt(0),
+					lastGeneratedHeight: 0,
+					isBanned: false,
+					pomHeights: [],
+					consecutiveMissedBlocks: 0,
+				}));
+			delegateAddresses = Array(103)
```

Comment:
You can also use, `Array.from({ length: 103 }, _ => getRandomBytes(20))`

Reply:
done